thewalrus/quantum/__init__.py

```diff
@@ -0,0 +1,202 @@
+    fidelity
+
+Details
+^^^^^^^
+
+.. autofunction::
+    pure_state_amplitude
+
+.. autofunction::
+    state_vector
+
+.. autofunction::
+    density_matrix_element
+
+.. autofunction::
+    density_matrix
+
+.. autofunction::
+    fock_tensor
+
+.. autofunction::
+    probabilities
+
+.. autofunction::
+    update_probabilities_with_loss
+
+.. autofunction::
+    update_probabilities_with_noise
+
+.. autofunction::
+    normal_ordered_expectation
+
+.. autofunction::
+    fidelity
+
+
+Utility functions
+-----------------
+
+.. autosummary::
+
+    reduced_gaussian
+    Xmat
+    Qmat
+    Covmat
+    Amat
+    complex_to_real_displacements
+    real_to_complex_displacements
+    find_scaling_adjacency_matrix
+    mean_number_of_clicks
+    mean_number_of_clicks_graph
+    find_scaling_adjacency_matrix_torontonian
+    gen_Qmat_from_graph
+    photon_number_mean
+    photon_number_mean_vector
+    photon_number_covar
+    photon_number_covmat
+    is_valid_cov
+    is_pure_cov
+    is_classical_cov
+    find_classical_subsystem
+    pure_state_distribution
+
+
+Details
+^^^^^^^
+"""
+
+from .fock_tensors import (
+    pure_state_amplitude,
+    state_vector,
+    density_matrix_element,
+    density_matrix,
+    fock_tensor,
+    probabilities,
+    loss_mat,
+    update_probabilities_with_loss,
+    update_probabilities_with_noise,
+    find_classical_subsystem,
+    _prefactor,
+)
+
+from .adjacency_matrices import (
+    find_scaling_adjacency_matrix,
+    find_scaling_adjacency_matrix_torontonian,
+    gen_Qmat_from_graph,
+)
+
+from .gaussian_state_tests import (
+    is_valid_cov,
+    is_pure_cov,
+    is_classical_cov,
+    fidelity,
+)
+
+from .conversions import (
+    reduced_gaussian,
+
+    Xmat,
+    Qmat,
+    Covmat,
+    Amat,
+    complex_to_real_displacements,
+    real_to_complex_displacements,
+)
+
+from .means_and_variances import (
+    photon_number_mean,
+    photon_number_mean_vector,
+    photon_number_covar,
+    photon_number_covmat,
+    photon_number_expectation,
+    photon_number_squared_expectation,
+    normal_ordered_expectation,
+    mean_number_of_clicks,
+    variance_number_of_clicks,
+    mean_number_of_clicks_graph,
+)
+
+from .photon_number_distributions import (
+    pure_state_distribution,
+    _squeezed_state_distribution,
+    _convolve_squeezed_state_distribution,
+)
+
+__all__ = [
+    "pure_state_amplitude",
+    "state_vector",
+    "density_matrix_element",
+    "density_matrix",
+    "fock_tensor",
+    "probabilities",
+    "update_probabilities_with_loss",
+    "update_probabilities_with_noise",
+    "normal_ordered_expectation",
+    "fidelity",
+]
+
+
+# old names for functions; remove in due time
+Means = real_to_complex_displacements
+Beta = complex_to_real_displacements
+prefactor = _prefactor
+gen_multi_mode_dist = _convolve_squeezed_state_distribution
+gen_single_mode_dist = _squeezed_state_distribution
+total_photon_num_dist_pure_state = pure_state_distribution
```

Comment:
Nice :+1:

I wonder if it is easy to raise a `DeprecationWarning` if a developer loads these functions with the old names... iirc, it is quite complicated for module-level variables.

Reply:
Seems like a good thing to have. I'll look into it.

Reply:
Added deprecation warnings for old names using a wrapper around the new functions. I couldn't think of any nicer way of doing this; but I think this works fairly well. :slightly_smiling_face: